### PR TITLE
Allow viewers to use oc status

### PR DESCRIPTION
**Pull request by deads2k**
Viewers don't have rights to view secrets, so the list from oc status fails.  This pull makes us tolerate forbidden listings and keep track of which types weren't retrieved.

Later, we'll be able to cull the list of warnings based on nodes we weren't able to retrieve.

@ironcladlou @fabianofranz ptal

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: pkg/cmd/server/origin/handlers.go [hunk 1]
```diff
@@ -84,6 +90,48 @@ func (c *MasterConfig) authorizationFilter(handler http.Handler) http.Handler {
...
+	if requestedCodec, err := klatest.InterfacesFor(apiVersion); err == nil {
+		codec = requestedCodec
+	}
+
+	formatted := &bytes.Buffer{}
+	output, err := codec.Encode(&forbiddenError.ErrStatus)
+	if err != nil {
+		fmt.Fprintf(formatted, "%s", forbiddenError.Error())
+	} else {
+		_ = json.Indent(formatted, output, "", "  ")
```
Comment (ironcladlou) [inline pkg/cmd/server/origin/handlers.go]: If you're going to eat the error, might as well simplify to just:

```go
json.Indent(formatted, output, "", "  ")
```

**Comment by ironcladlou** [comment]
Only had time for a quick look but so far LGTM

File: pkg/cmd/server/origin/handlers.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/origin/handlers.go]: > If you're going to eat the error, might as well simplify to just:

done.

**Comment by fabianofranz** [comment]
Reviewed, LGTM

**Comment by deads2k** [comment]
[merge]

**Pull request merged**